crates/lang_srv/src/analysis.rs

```diff
@@ -362,38 +461,57 @@ impl AnalyzedDocument {
         Some(GotoDefinitionResponse::Scalar(self.location(range)))
     }
 
-    pub fn format(&self) -> Option<Vec<TextEdit>> {
-        let source = &self.source;
-        let arena = &Bump::new();
-
-        let ast = Ast::parse(arena, source).ok()?;
-        let fmt = ast.fmt();
-
-        if source == fmt.as_str() {
-            None
-        } else {
-            let range = self.whole_document_range();
-            let text_edit = TextEdit::new(range, fmt.to_string().to_string());
-            Some(vec![text_edit])
-        }
+    pub(crate) fn module_url(&self, module_id: ModuleId) -> Option<Url> {
+        self.module()?.module_id_to_url.get(&module_id).cloned()
     }
 
-    pub fn semantic_tokens(&self) -> Option<SemanticTokensResult> {
-        let source = &self.source;
-        let arena = &Bump::new();
-
-        let ast = Ast::parse(arena, source).ok()?;
-        let tokens = ast.semantic_tokens();
+    pub fn completion_items(
+        &self,
+        position: Position,
+        latest_doc: &DocInfo,
+    ) -> Option<Vec<CompletionItem>> {
+        let symbol_prefix = latest_doc.get_prefix_at_position(position);
+        eprintln!(
+            "starting to get completion items for prefix: {:?} docVersion:{:?}",
+            symbol_prefix, latest_doc.version
+        );
```

Comment:
I am using a logging library rather than roc_tracing because I want it to be enabled in release builds. We do want people to be able to record a log of the lang_server for debugging issues. If you'd prefer I adjust roc_tracing and use that I'm happy to